lib/convection/control/stack.rb

```diff
@@ -586,12 +582,39 @@ def get_events(pages = nil, stack_name = id)
         end
       rescue Aws::CloudFormation::Errors::ValidationError # Stack does not exist
       rescue Aws::EC2::Errors::RequestExpired
-        raise unless @credential_error_count <= @credential_error_max_retries
+        raise if reached_max_credential_retries?
+
+        wait_for_credential_refresh
+
+        reload_cf_client
+        retry
+      end
+
+      def client_options
+        client_options = { region: @region }
+        client_options[:credentials] = @credentials unless @credentials.nil?
 
+        client_options
+      end
+
+      def reload_cf_client
+        Aws.shared_config.fresh if @credentials.nil?
```

Comment:
@athompson-r7 nice catch. I looked more into using public APIs for doing this and, as far as I can tell, the only way to reload the credentials (not the actual file) would be by using [`Aws::AssumeRoleCredentials`](https://github.com/aws/aws-sdk-ruby/blob/ef9f2d3655c4bacb3adff580a6b295df510f78fe/aws-sdk-core/lib/aws-sdk-core/assume_role_credentials.rb) unfortunately. 🙍

This could be something we configured but we'd need a set of persistent credentials that allowed us to the use `Amazon::STS::Client` API to refresh credentials when we're nearing expiry.

That may sound familiar since we do the same in Awsaml and the work in this PR as well as #174 was to enable reloading the credentials Awsaml re-wrote to the credentials file. 😥